src/internal/connector/exchange/service_iterators.go

```diff
@@ -63,8 +64,8 @@ func filterContainersAndFillCollections(
 	}
 
 	for _, c := range resolver.Items() {
-		if ctrlOpts.FailFast && errs != nil {
-			return errs
+		if errs.Failed() {
+			return errs.Err()
```

Comment:
a bit concerned here too about only returning one of possibly several errors. Seems like either this function should no longer return an error and rely on the in/out param `errs` or should return all errors in `errs`

Reply:
I feel like there's some misunderstanding of the fault package behavior we need to clear up.  We can't lose or hide errors, except by ignoring their population when we handle errors upstream.  What we're doing here isn't deciding _what_ errors to return.  It's only a consideration of _when_ to stop processing and return.